### PR TITLE
jenkins: model replay report in PR

### PR DESCRIPTION
**Pull request by maxime-desroches**
~need to redeploy jenkins to get the new PAT~

example: https://github.com/maxime-desroches/test_rewrite/pull/24

### CODE REVIEW

File: selfdrive/test/process_replay/model_replay.py [hunk 1]
```diff
@@ -19,10 +23,55 @@
 NO_MODEL = "NO_MODEL" in os.environ
 SEND_EXTRA_INPUTS = bool(int(os.getenv("SEND_EXTRA_INPUTS", "0")))
 
+DATA_TOKEN=os.getenv("CI_ARTIFACTS_TOKEN","")
```
Comment (adeebshihadeh) [inline selfdrive/test/process_replay/model_replay.py:26]:
```suggestion
DATA_TOKEN = os.getenv("CI_ARTIFACTS_TOKEN","")
```

File: tools/lib/github_utils.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+      case HTTPMethod.GET:
+        return requests.get(path, headers=headers)
+      case HTTPMethod.PUT:
+        return requests.put(path, headers=headers, data=data)
+      case HTTPMethod.POST:
+        return requests.post(path, headers=headers, data=data)
+      case HTTPMethod.PATCH:
+        return requests.patch(path, headers=headers, data=data)
+      case _:
+        raise NotImplementedError()
```
Comment (adeebshihadeh) [inline tools/lib/github_utils.py:39]: can't you just use `requests.request("GET"...` here?

File: tools/lib/github_utils.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+      sha = f'"sha":"{sha}",' if sha else ''
+
+      data = f'{{"message":"uploading {file_name}", \
+                    "branch":"{bucket}", \
+                    "committer":{{"name":"Vehicle Researcher", "email": "user@comma.ai"}}, \
+                    {sha} \
+                    "content":"{encoded}"}}'
+      github_path = f"contents/{file_name}"
+      if not self.api_call(github_path, data=data, method=HTTPMethod.PUT, data_call=True).ok:
+        raise Exception(f"Error uploading {file_name} to {bucket}")
```
Comment (adeebshihadeh) [inline tools/lib/github_utils.py:56]: I'd just assert in `api_call` instead of doing it in all its consumers

**Comment by commaci-public** [comment]
test comment

**Pull request merged**